test/users_finish_test.rb

```diff
@@ -71,29 +74,87 @@
         let(:autoinst) { false }
 
         before do
-          allow(subject).to receive(:setup_all_users).and_return(user_added)
+          # Mocking to avoid to write to the system
+          allow_any_instance_of(Y2Users::Linux::Writer).to receive(:write).and_return(issues)
+
+          # Mocking to avoid to read from the system
+          allow(Y2Users::ConfigManager.instance).to receive(:system).and_return(system_config)
+
+          allow(Y2Users::ConfigManager.instance).to receive(:config).with(:system)
+            .and_return(system_config)
+
+          allow(Y2Users::ConfigManager.instance).to receive(:config).with(:target)
+            .and_return(target_config)
+
+          allow(Yast::UsersSimple).to receive(:AutologinUsed).and_return(autologin)
+          allow(Yast::UsersSimple).to receive(:GetAutologinUser).and_return(autologin_user)
+        end
+
+        let(:system_config) { Y2Users::Config.new }
+
+        let(:target_config) { Y2Users::Config.new }
+
+        let(:autologin) { nil }
+
+        let(:autologin_user) { nil }
+
+        let(:writer) { instance_double(Y2Users::Linux::Writer, write: issues) }
+
+        let(:issues) { [] }
+
+        it "calls the linux writer with the expected configs" do
+          expect(Y2Users::Linux::Writer).to receive(:new).with(target_config, system_config)
+            .and_return(writer)
+
+          expect(writer).to receive(:write)
+
+          subject.run
+        end
+
+        context "when the writer does not report issues" do
+          let(:issues) { [] }
+
+          it "does not report errors to the user" do
+            expect(Yast::Report).to_not receive(:Error)
+
+            subject.run
+          end
+        end
+
+        context "when the writer reports issues" do
+          let(:issues) { [issue1, issue2] }
+
+          let(:issue1) { Y2Issues::Issue.new("error 1") }
+
+          let(:issue2) { Y2Issues::Issue.new("error 2") }
+
+          it "reports all errors to the user" do
+            expect(Yast::Report).to receive(:Error).with("error 1\n\nerror 2")
+
+            subject.run
+          end
         end
 
-        context "when a new user is added" do
-          let(:user_added) { true }
+        context "when an user is configured for auto-login" do
```

Comment:
NP: I think it is [_**a** user_](https://english.stackexchange.com/questions/105116/is-it-a-user-or-an-user). And I think now that probably I wrote _an user_ in the past and/or did not catch them in previous reviews.